torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -191,25 +191,27 @@ void IRPrinter::visit(const CompareSelectPtr& v) {
   withParens(v->ret_val2());
 }
 
-static void formatFPSuffix(std::ostream& os, double v) {
-  os << (v == std::ceil(v) ? ".0" : "");
+static void formatFPSuffix(std::ostream& os, double v, bool flag) {
+  os << (flag && v == std::ceil(v) ? ".0" : "");
 }
 
 template <typename T>
-static void formatFPSuffix(std::ostream& os, T v) {
-  os << (v == std::ceil(v) ? ".f" : "f");
+static void formatFPSuffix(std::ostream& os, T v, bool flag) {
+  os << (flag && v == std::ceil(v) ? ".f" : "f");
 }
 
 template <typename T, std::enable_if_t<std::is_floating_point_v<T>>* = nullptr>
 static void formatImm(std::ostream& os, T v) {
   const int precision = 16;
+  constexpr T lower_bound = static_cast<T>(-std::pow(10, precision));
+  constexpr T upper_bound = -lower_bound ;
```

Comment:
@pobin6 
```suggestion
  const T lower_bound = static_cast<T>(-std::pow(10, precision));
  const T upper_bound = -lower_bound ;
```

Reply:
The failures preventing this PR from being merged are all
`constexpr variable 'upper_bound' must be initialized by a constant expression`
(https://github.com/pytorch/pytorch/pull/140002#issuecomment-2462093346).

According to https://en.cppreference.com/w/cpp/numeric/math/pow, `std::pow` won't be `constexpr` until C++26. I don't know why your sample works. The failures are with Clang so maybe GCC has a non-standard extension

Reply:
Confirming the compiler difference:
```sh
james@work:~/code$ g++ main.cpp -std=c++11
james@work:~/code$ ./a.out
-10000000272564224.000000
james@work:~/code$ clang-19 main.cpp -std=c++11
main.cpp:3:17: error: constexpr variable 'lower_bound' must be initialized by a constant expression
    3 | constexpr float lower_bound = static_cast<float>(-std::pow(10, 16));
      |                 ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/cmath:1073:14: note: non-constexpr function 'pow' cannot be used in a constant expression
 1073 |       return pow(__type(__x), __type(__y));
      |              ^
main.cpp:3:51: note: in call to 'pow<int, int>(10, 16)'
    3 | constexpr float lower_bound = static_cast<float>(-std::pow(10, 16));
      |                                                   ^~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:140:17: note: declared here
  140 | __MATHCALL_VEC (pow,, (_Mdouble_ __x, _Mdouble_ __y));
      |                 ^
1 error generated.
```